### PR TITLE
Shut down agent after each test

### PR DESCRIPTION
**Pull request by odisseus**
Since the `CodyIntegrationTextFixture` starts a fresh agent for every test case, we need to shut down those agents during cleanup. Otherwise the leftover agent processes clog the RAM (each Node process takes 50-250 MB), and the JVM threads which listen to them keep running forever.

## Test plan

Manually run the stability check in CI
[1000 repetitions](https://github.com/sourcegraph/jetbrains/actions/runs/10073107716) (5000 individual test cases): 2 hours 40 minutes, 118 failures

### CODE REVIEW

**Review by mkondratek — APPROVED** [review]
lgtm

File: src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTextFixture.kt [hunk 1]
```diff
@@ -49,6 +49,13 @@ open class CodyIntegrationTextFixture : BasePlatformTestCase() {
           logger.warn("Error shutting down session", x)
         }
       }
+      CodyAgentService.getInstance(myFixture.project).apply {
+        try {
+          dispose()
```
Comment (pkukielka) [inline src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTextFixture.kt]: This is not shutting down agent, but finishing current fixup session.
IMHO it makes sense to wait for such finish, but if it is not able to finish itself in a given time window it should be a test failure.
So instead of disposing it we should wait and check there.

File: src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTextFixture.kt [hunk 2]
```diff
@@ -259,7 +266,7 @@ open class CodyIntegrationTextFixture : BasePlatformTestCase() {
   }
 
   companion object {
-    const val ASYNC_WAIT_TIMEOUT_SECONDS = 10L
+    const val ASYNC_WAIT_TIMEOUT_SECONDS = 30L
```
Comment (pkukielka) [inline src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTextFixture.kt]: Why is that needed? In the replay mode whole test should finish in well under 10s, not to mention single call.
Or maybe is it needed for the recording mode?
If so, maybe we should have configurable timeouts in the gradle config?

Comment (odisseus) [reply src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTextFixture.kt]: Reverted
